mux/src/tmux_commands.rs

```diff
@@ -255,14 +674,267 @@ impl TmuxCommand for ListAllPanes {
                 pane_height,
                 pane_left,
                 pane_top,
+                pane_active,
+            });
+        }
+
+        log::debug!("panes in domain_id {}: {:?}", domain_id, items);
+        let mux = Mux::get();
+        if let Some(domain) = mux.get_domain(domain_id) {
+            if let Some(tmux_domain) = domain.downcast_ref::<TmuxDomain>() {
+                if !self.prune {
+                    return tmux_domain.inner.sync_pane_state(&items);
+                } else {
+                    return tmux_domain
+                        .inner
+                        .remove_detached_pane(self.window_id, &pane_set);
+                }
+            }
+        }
+        anyhow::bail!("Tmux domain lost");
+    }
+}
+
+fn parse_pane(layout: &str) -> anyhow::Result<PaneItem> {
+    let re_pane = Regex::new(r"^,?(\d+)x(\d+),(\d+),(\d+)(,(\d+))?").unwrap();
```

Comment:
With the Regex stuff, I'm a bit conflicted.  The intent of the `tmux.pest` file is that it encapsulates the parsing of the tmux protocol in a well-structured way that is a bit more rigorous than a bunch of regular expressions.

I am pragmatic though: if there is a good reason why it is not possible to express things in pest, then we can accommodate the regex.  If we need to go the regex route, then I'd recommend using a LazyLock to compile each regex once:

```rust
static RE_PANE: LazyLock<Regex> = LazyLock::new(|| Regex::new("something").unwrap());
```

(I'm cool with an unwrap in this situation: if the regex doesn't parse, that's a showstopping problem)